flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -176,23 +176,20 @@ public KubernetesClient getKubernetesClient() {
 
     @Override
     public void submitApplicationCluster(
-            JobSpec jobSpec, Configuration conf, boolean requireHaMetadata) throws Exception {
+            JobSpec jobSpec, Configuration conf, boolean requireHaMetadata, boolean requireScale)
+            throws Exception {
         LOG.info(
                 "Deploying application cluster{}",
                 requireHaMetadata ? " requiring last-state from HA metadata" : "");
 
-        // If Kubernetes or Zookeeper HA are activated, delete the job graph in HA storage so that
-        // the newly changed job config (e.g. parallelism) could take effect
-        if (FlinkUtils.isKubernetesHAActivated(conf)) {
-            final String clusterId = conf.get(KubernetesConfigOptions.CLUSTER_ID);
-            final String namespace = conf.get(KubernetesConfigOptions.NAMESPACE);
-            FlinkUtils.deleteJobGraphInKubernetesHA(clusterId, namespace, kubernetesClient);
-        } else if (FlinkUtils.isZookeeperHAActivated(conf)) {
-            FlinkUtils.deleteJobGraphInZookeeperHA(conf);
-        }
-
         if (requireHaMetadata) {
             validateHaMetadataExists(conf);
+            LOG.info(
+                    "Deleting job graph from HA metadata-{}",
```

Comment:
I think this logging is off...
Can we simply log `Deleting ....` or `Not deleting ...`?